kafkahook/kafkahook.go

```diff
@@ -149,5 +171,13 @@ func (h *KafkaHook) SetDebug(set bool) {
 
 // Close the kakfa producer and flush any remaining logs
 func (h *KafkaHook) Close() error {
-	return h.producer.Close()
+	var err error
+	if h.once != nil {
+		h.once.Do(func() {
```

Comment:
Are you using once so that Close can be called many times without panicking? That does not really necessary since Logrus will call it only once. But I do not mind. You can keep it if you won't just make h.once a value rather than a pointer.

Reply:
Call me overly cautious, I've been bitten by strange double close issues before. Upstream libraries don't always do what they are supposed, and you don't find out until you have upgraded to the latest version of the upstream library and your app panic in production.